deploy/vagrant/bin/spot_request.py

```diff
@@ -203,12 +203,16 @@ def main(args):
         info('getting instance IPs...')
         for r in requests:
             instance_id = r.instance_id
-            ip = conn.get_only_instances([instance_id])[0].ip_address
+            info('waiting for ip to be allocated to the machine')
+            ip = None
+            while ip is None:
+                ip = conn.get_only_instances([instance_id])[0].ip_address
```

Comment:
Does this call always return None the first time? If not, maybe only start waiting/sleeping after the first time.